Fix theme for Slack 3.2.0

Nockiro opened the pull request:
Since the codes and workarounds in #48 do work but don't include the theme colors anymore, this should fix most of the broken areas taking the theme variables into account.

Nockiro committed:
- add searchbox to custom.css

Nockiro committed:
- design fixes for slack 3.2.0

Nockiro committed:
- more design fixes

Nockiro committed:
- more slack 3.x adjustments

Nockiro committed:
- more slack 3.x adjustments

Nockiro committed:
- more slack 3.x adjustments

Nockiro committed:
-- more slack 3.x adjustments (modal transitions, more menus)

Nockiro committed:
- more minor fixes for 3.2.0

Nockiro committed:
- fixed scrollbar

Nockiro committed:
- changes to details tab

Nockiro committed:
# fix for channels with limited history

Nockiro committed:
# fixed message text color in chat and search modal

Nockiro committed:
- fixed a few more small things like special buttons, modals and emojis

Nockiro committed:
- fixed more exotic things

Nockiro committed:
# file container fixes

Nockiro committed:
# file container fix fixes

Nockiro committed:
# fixes for several smaller elements but also the background "shadow" on "new message" banner

Nockiro committed:
- adjusted colors in sidebars to the same scheme
- fixed channel browser, some hovering and button styles

Nockiro committed:
- additions to ef89e3427d15da3471ab2704c04a8330f18f4d7d, app browser and some more button fixes

Nockiro committed:
- fixed sidebar "jumper", a few hovers and "whats new" tab

Nockiro committed:
- slight adjustments of old modifications and minor bugs

Nockiro committed:
- button and sidebar adjustments

Nockiro committed:
- more minor fixes a.o. for search modal (handling widget-/slack-black-theme#61 )

Nockiro committed:
- adjusted message group hovers

eduardonunesp commented:
Looks like the threads input text aren't following the theme. https://cl.ly/37210F1b0E3U

eduardonunesp commented:
User local time isn't appearing correctly 
![image](https://user-images.githubusercontent.com/582516/43405617-a039a008-93f0-11e8-8e49-25586c955d98.png)

Nockiro committed:
# fixed "new thread" to follow design in "All Threads" ( https://github.com/widget-/slack-black-theme/pull/60#issuecomment-408880170 )
# fixed local time ( https://github.com/widget-/slack-black-theme/pull/60#issuecomment-408895491 )

intere commented:
@widget- any chance you can get this tested / merged in?  For now I've just updated the reference JS to point to @Nockiro's fixes.  It's looking good on my systems.

intere commented:
@widget- any chance you could take a look at this?

parga commented:
@widget- can we have this merged ?

neemzy commented:
@widget- Hello! Is there any chance you might take a look at this? Thanks a lot! :)

SystemDisc commented:
@Nockiro There are still a couple places in All Threads and the Thread Sidebar on the right where the theme is broken, even in the latest commit in this PR.

Thanks so much for fixing pretty much everything, though.

ssoroka commented:
Until this is merged, I forked the repo and merged Nockiro's two fix branches. ~you can use this patched version at https://raw.githubusercontent.com/ssoroka/slack-black-theme/master/custom.css~

Edit: Or use Nockiro's master branch in the next post; he seems more likely to keep it up to date anyway. :)

Nockiro commented:
@SystemDisc I can't find any, if you could take a screenshot of the glitches, it would maybe help me to see them - thanks for reporting, though!
@ssoroka Actually the current master branch (https://raw.githubusercontent.com/Nockiro/slack-black-theme/master/custom.css) should have everything you need, I just keep the two fix- branches separately to (if the time should ever come) merge the changes into this repository over a new pull request - I just didn't want to open a second pull request for the >3.3 changes as long as the first one isn't even merged yet.

treycedits commented:
@Nockiro thanks for all the fixes! Unfortunately I tried using the master branch but when opening Slack it just infinitely shows the loading screen and 'Connecting to Slack'. Anything I'm doing wrong?

SystemDisc commented:
@Nockiro I'm using the most recent commit in this PR. Here's what I see:

<img width="644" alt="screen shot 2019-02-05 at 11 35 37 am" src="https://user-images.githubusercontent.com/1561860/52299652-d59a1e00-293a-11e9-98e0-462dea2a7f5a.png">  
<img width="1792" alt="screen shot 2019-02-05 at 11 35 01 am" src="https://user-images.githubusercontent.com/1561860/52299661-dc289580-293a-11e9-8035-fb9e44b1f696.png">

elv1n commented:
I'm supporting https://github.com/elv1n/slack-dark-mojave-theme to be up to date with slack updates

metaskills commented:
Changed my CSS line to `const cssPath = 'https://raw.githubusercontent.com/widget-/slack-black-theme/3ea2efdfb96ccc91549837ab237d57104181bbf8/custom.css';` and this worked well.

SystemDisc commented:
I've switched to using https://github.com/elv1n/slack-dark-mojave-theme. As of today, there's an issue with the "All Threads" view, but it seems a lot easier to use and seems to get updates more frequently - at least for now.
